sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -18,10 +29,322 @@ internal class RadarVerificationManager(
+
+                                            Radar.beaconManager.rangeBeaconUUIDs(
+                                                uuids,
+                                                uids,
+                                                false,
+                                                object : Radar.RadarBeaconCallback {
+                                                    override fun onComplete(
+                                                        status: Radar.RadarStatus,
+                                                        beacons: Array<RadarBeacon>?
+                                                    ) {
+                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
+                                                            callTrackApi(null)
+
+                                                            return
+                                                        }
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else if (beacons != null) {
+                                            Radar.beaconManager.startMonitoringBeacons(beacons)
+
+                                            Radar.beaconManager.rangeBeacons(
+                                                beacons,
+                                                false,
+                                                object : Radar.RadarBeaconCallback {
+                                                    override fun onComplete(
+                                                        status: Radar.RadarStatus,
+                                                        beacons: Array<RadarBeacon>?
+                                                    ) {
+                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
+                                                            callTrackApi(null)
+
+                                                            return
+                                                        }
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else {
+                                            callTrackApi(null)
+                                        }
+                                    }
+                                },
+                                false
+                            )
+                        } else {
+                            callTrackApi(null)
+                        }
+                    }
+                }
+            })
+    }
+
+    fun startTrackingVerified(token: Boolean, interval: Int, beacons: Boolean) {
```

Comment:
Something I'm gonna noodle on / want to think through / test: how does this behave across the app lifecycle. Do we want anything to restart this after being force-killed?

Reply:
We'll just tell developers to do `Radar.initialize()` and `Radar.startTrackingVerified()` on app init. Everything will die on force quit and it's fine if it's up to the developer to restart it. We should be checking and avoiding recreating timers and such